exchanges/request/limit.go

```diff
@@ -133,44 +126,59 @@ func (r *Requester) GetRateLimiterDefinitions() RateLimitDefinitions {
 	return r.limiter
 }
 
-// RateLimit is a function that will rate limit a request based on the rate
-// limiter provided. It will return an error if the context is cancelled or
-// deadline exceeded.
-func RateLimit(ctx context.Context, rateLimiter *RateLimiterWithWeight) error {
-	if rateLimiter == nil {
-		return errSpecificRateLimiterIsNil
+// RateLimit is a function that will rate limit a request based on the rate limiter provided. It will return an error if
+// the context is cancelled, deadline exceeded or if no delay is permitted via the context and a delay is required.
```

Comment:
🐒 Optional nitpick:
* This is a method, so "on the rate limiter provided" is implied
* Erroring on context cancelled or deadline exceeded is implicit always when using contexts, so we can remove those comments. And conversely: Any function *not* respecting the context given to it is a bug 🐛 
* I've attempted to get the wording right for the verb for this function ... `throttles` seemed kinda-okay, but I've gone for `delays`
```suggestion
// RateLimit throttles a request based on weight, delaying the request 
// Errors if no delay is permitted via the context and a delay is required
```